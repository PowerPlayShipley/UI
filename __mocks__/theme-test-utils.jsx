// TODO: Fix this??

import * as React from 'react'
import userEvent from '@testing-library/user-event'
import { render as rtlRender } from '@testing-library/react'

import { ThemeProvider } from "styled-components";

import theme from '../src/themes'
import GlobalStyle from "../src/components/GlobalStyle";

function renderWithTheme(ui, { ...options } = {}) {
  const Wrapper = ({ children }) => (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
      <div id='modal-root' />
    </ThemeProvider>
  )

  return rtlRender(ui, { wrapper: Wrapper, ...options })
}

// THE BELLOW COME FROM `https://github.com/styled-components/styled-components/blob/0aa3170c255a49cd41c3fbeb2b8051b5d132f229/src/test/utils.js`

const stripComments = (str) => str.replace(/\/\*.*?\*\/\n?/g, '');

const stripWhitespace = (str) => str
  .trim()
  .replace(/([;\{\}])/g, '$1  ')
  .replace(/\s+/g, ' ');

const getCSS = (scope) =>
  Array.from(scope.querySelectorAll('style'))
    .map(tag => tag.innerHTML)
    .join('\n')
    .replace(/ {/g, '{')
    .replace(/:\s+/g, ':')
    .replace(/:\s+;/g, ':;');

function expectCSSMatches ( _expectation, opts = { ignoreWhitespace: true }) {
  // NOTE: This should normalise both CSS strings to make irrelevant mismatches less likely
  const expectation = _expectation
    .replace(/ {/g, '{')
    .replace(/:\s+/g, ':')
    .replace(/:\s+;/g, ':;');

  const css = getCSS(document);

  if (opts.ignoreWhitespace) {
    const stripped = stripWhitespace(stripComments(css));
    expect(stripped).toEqual(stripWhitespace(expectation));
    return stripped;
  } else {
    expect(css).toEqual(expectation);
    return css;
  }
};

export * from '@testing-library/react';

export { renderWithTheme, expectCSSMatches };
