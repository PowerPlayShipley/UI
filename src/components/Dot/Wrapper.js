import styled from "styled-components";
import PropTypes from "prop-types";

const Wrapper = styled.span`
  height: ${({size}) => size};
  width: ${({size}) => size};
  
  background-color: ${({color}) => color};
  
  border-radius: 50%;
  display: inline-block;
`

Wrapper.propTypes = {
  size: PropTypes.string,
  color: PropTypes.string
}

export default Wrapper
