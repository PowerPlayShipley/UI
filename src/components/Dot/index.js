import React from 'react'
import clsx from 'clsx'
import PropTypes from 'prop-types'
import { useTheme } from "styled-components";

import Wrapper from './Wrapper'

const sizeToScaleSwitch = (size, scaleMap) => {
  return size === 'sm' ? scaleMap['size-1'] : size === 'md' ? scaleMap['size-2'] : scaleMap['size-3']
}

function Dot ({ className, size = 'md', color = '#bbb', ...rest }) {
  const { main: { sizes: { scale } } } = useTheme()
  return <Wrapper className={clsx(className)} size={sizeToScaleSwitch(size, scale)} color={color} {...rest} />
}

Dot.propTypes = {
  size: PropTypes.oneOf([
    'sm', 'md', 'lg'
  ]),
  color: PropTypes.string
}

export default Dot
