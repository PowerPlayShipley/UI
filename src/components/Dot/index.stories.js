import Dot from "./index";

export default {
  title: 'Dot',
  component: Dot
}

const Template = (args) => (<Dot {...args} />);

export const Default = Template.bind({});
Default.args = {
  size: 'md',
  color: 'green'
}
