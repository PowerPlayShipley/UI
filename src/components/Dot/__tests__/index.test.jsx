/**
 * @jest-environment jsdom
 */

import React from 'react'
import { renderWithTheme } from '../../../../__mocks__/theme-test-utils'

import Dot from "../index";

describe('Display Dot', () => {
  it('should render dot', function () {
    const { container } = renderWithTheme(<Dot />)
    // Very strange but works
    expect(container).toContainHTML('</span>')
  });

  it('should create valid snapshot', () => {
    const { container } = renderWithTheme(<Dot />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
