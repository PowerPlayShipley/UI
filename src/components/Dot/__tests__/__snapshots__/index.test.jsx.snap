// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Dot should create valid snapshot 1`] = `
.c0 {
  height: 20px;
  width: 20px;
  background-color: #bbb;
  border-radius: 50%;
  display: inline-block;
}

<span
  class="c0"
  color="#bbb"
/>
`;
